cmd/registry/controller/controller_test.go

```diff
@@ -774,6 +778,31 @@ func TestMultipleEntitiesArtifacts(t *testing.T) {
 			}
 			actions := ProcessManifest(ctx, lister, projectID, manifest, 10)
 
+			if test.generatedResourceRequiresRevision {
```

Comment:
I am not sure if we need this flag, wouldn't all generatedResource should include revision IDs in them? Since we are saying that we want to carry through a spec revision till the action.

IMO every reference of a spec should have a revision ID to it, we can't rely on `projects/demo/locations/apis/petstore/versions/v1/specs/openapi.yaml` anywhere because the latest revision might change anytime while the controller is processing the sate and taking actions. Every spec should include the revision ID in the controller.